### PR TITLE
feat(eap): add support for virtual columns in the protobuf

### PR DESCRIPTION
**Pull request by volokluev**
There is a requirement to have the ability to search by virtual columns. Remove the AttributeKeyTransformContext which special cased proejct id and make it generic for any desired virtual column

### CODE REVIEW

**Comment by colin-sentry** [comment]
how would this work for the uint case, everything gets converted to strings?

**Comment by volokluev** [comment]
> how would this work for the uint case, everything gets converted to strings?

That's what I was thinking. I mostly did this because I can't do a `oneOf` map in protobuf. But if you have another solution that allows specifying both I'm happy to hear it

File: proto/sentry_protos/snuba/v1alpha/trace_item_attribute.proto [hunk 1]
```diff
@@ -18,9 +18,14 @@ message AttributeKey {
...
-// Special cases for attributes, in particular things like project name
-message AttributeKeyTransformContext {
-  map<uint64, string> project_ids_to_names = 1;
+// custom mappings of column values which are not present in EAP storage
+// for example, project name is changeable by the user and not stored in EAP
+// but sorting by the project name may be desired. This struct carries that mapping
+// data
+message VirtualColumnContext {
+  string from_column_name = 1;
+  string to_column_name = 2;
```
Comment (Zylphrex) [inline proto/sentry_protos/snuba/v1alpha/trace_item_attribute.proto:27]: Few questions here
- Does this mean `to_column_name` will be the name of the virtual column?
- Is this name constrained anywhere or is it an arbitrary string?
- Will this guarantee the `to_column_name` will be the name of a column in the response?
- Will this name be accessible in other parts of the query? So think `count_unique(project)` will get the project names substituted in automatically? (not saying we'll do this, just want to understand the semantics of this context).

Comment (volokluev) [reply proto/sentry_protos/snuba/v1alpha/trace_item_attribute.proto:27]: > Does this mean to_column_name will be the name of the virtual column?

Yes 

> Is this name constrained anywhere or is it an arbitrary string?

Whatever you want bb

> Will this guarantee the to_column_name will be the name of a column in the response?

Yes

> Will this name be accessible in other parts of the query? So think count_unique(project) will get the project names substituted in automatically? (not saying we'll do this, just want to understand the semantics of this context).

This is a good question. I would say for now these virtual columns can be used to order by and in the result set. aggregations on virtual columns not supported

**Review by Zylphrex — APPROVED** [review]

**Pull request merged**